### PR TITLE
Hidden files caused build/test to fail: only look at directories in test files

### PR DESCRIPTION
**Pull request by jdelbick**
## Description

This issue came up when there was a hidden file in the test cases folder: 
```
  test/asset-compute/
        worker/
        .DS_store
```

The issue did not come up when there was a hidden file in the worker actions folder:
```
actions/
        worker/
        .DS_store
```
(this is actually a common case that happens because sometimes our worker code is at the root of the project which also contains hidden files like .env. Test cases already covered this):
```
project/
    worker.js
    .env
    test/
    ...
```
The solution was to only look at directories when a list of actions in the test/asset-compute/*

Fixes # https://jira.corp.adobe.com/browse/NUI-771
bug:
Screen Shot 2020-08-26 at 5 24 49 PM
with fix:
Screen Shot 2020-08-26 at 5 25 53 PM

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test-projects/test-hidden-file/test/asset-compute/worker/.test [hunk 1]
```diff
@@ -0,0 +1 @@
+hello
```
Comment (alexkli) [inline test-projects/test-hidden-file/test/asset-compute/worker/.test:1]: You could have "Bad guy" in here ;-)

Comment (alexkli) [reply test-projects/test-hidden-file/test/asset-compute/worker/.test:1]: Wait, shouldn't this be one level up, inside the `asset-compute` folder? As the original error is when that includes an extra file or directory that is not an action inside the manifest.yml?

And maybe use `.DS_Store` as name to closely reproduce the original issue.

File: src/commands/asset-compute/test-worker.js [hunk 2]
```diff
@@ -78,8 +78,21 @@ class TestWorkerCommand extends BaseCommand {
...
         //     workerB/
-
-        return fs.existsSync(TEST_DIR) ? fs.readdirSync(TEST_DIR) : [];
+        if (fs.existsSync(TEST_DIR)) {
+            const tests = fs.readdirSync(TEST_DIR);
+            // eslint-disable-next-line array-callback-return
+            return tests.filter(test => {
+                const testPath = path.resolve(TEST_DIR, test);
+                try {
+                    fs.readdirSync(testPath);
```
Comment (alexkli) [inline src/commands/asset-compute/test-worker.js:87]: Node has apis to to check for directories that doesn't require try/catch, for example `lstatSync(path).isDirectory()`.

But there is an even better way - from https://stackoverflow.com/a/24594123/2709:

```
fs.readdirSync(TEST_DIR, { withFileTypes: true })
    .filter(entry => entry.isDirectory())
    .map(entry => entry.name)
```

File: test-projects/test-hidden-file/test/asset-compute/worker/.test (hunk 1 above)
Comment (jdelbick) [reply test-projects/test-hidden-file/test/asset-compute/worker/.test:1]: oops yeah meant to put this one level up. and I'll name it .DS_Store so it makes more sense

**Pull request merged**